### PR TITLE
Pass empty tags on run creation, return tags

### PR DESCRIPTION
**Pull request by tomas-milata**
At the moment, the backend requires `tags` to be passed in the request to create a run. This fixes it by passing empty tags for now.

The next step is to pass through tags from mlflow-faculty to this client.

Also, tag schema was added to the run schema (used when fetching runs).

### CODE REVIEW

File: faculty/clients/experiment.py [hunk 1]
```diff
@@ -203,6 +222,7 @@ def create_run(
             The location of the artifact repository to use for this run.
             If omitted, the value of `artifact_location` for the experiment
             will be used.
+        tags: dict[str, str]
```
Comment (acroz) [inline faculty/clients/experiment.py]: Shouldn't this be `List[Tag]`?

File: tests/clients/test_experiment.py [hunk 2]
```diff
@@ -121,6 +128,11 @@
 }
 
 
+def test_tag_schema():
```
Comment (acroz) [inline tests/clients/test_experiment.py]: We should probably test the `.dump()` on this as well as it's being used for serialisation.

File: tests/clients/test_experiment.py [hunk 3]
```diff
@@ -121,6 +128,11 @@
...
+    data = TagSchema().load(TAG_BODY)
+    assert data == TAG
+
+
 def test_experiment_schema():
```
Comment (acroz) [inline tests/clients/test_experiment.py]: Can we also test the case where non-empty tags are passed?

File: faculty/clients/experiment.py [hunk 4]
```diff
@@ -188,7 +202,12 @@ def list(self, project_id):
         return self._get(endpoint, ExperimentSchema(many=True))
 
     def create_run(
-        self, project_id, experiment_id, started_at, artifact_location=None
+        self,
+        project_id,
+        experiment_id,
+        started_at,
+        artifact_location=None,
+        tags=[],
```
Comment (acroz) [inline faculty/clients/experiment.py]: Having mutable objects as default arguments is discouraged in Python for reasons I'd rather not type out (can explain in person). Recommend having `tags=None` here and then in the method have:

```python
if tags is None:
    tags = []
```

File: faculty/clients/experiment.py (hunk 1 above)
Comment (tomas-milata) [reply faculty/clients/experiment.py]: yes, thanks

File: tests/clients/test_experiment.py (hunk 3 above)
Comment (tomas-milata) [reply tests/clients/test_experiment.py]: I think this is now covered by the tests that Maitiu added and the one that I added.

File: tests/clients/test_experiment.py [hunk 5]
```diff
@@ -216,6 +229,11 @@ def test_experiment_run_tag_schema():
     assert data == TAG
 
 
+def test_experiment_run_tag_schema_dump():
```
Comment (acroz) [inline tests/clients/test_experiment.py]: This should probably just be `test_tag_schema_dump`.

Usually test names are in the format `test_foo_bar_method_name` when testing `FooBar.method_name`. If other tests break this we should fix. (Exceptions are usually made for schemas only used to convert in one direction, as is the case in most client modules in this library).

Comment (tomas-milata) [reply tests/clients/test_experiment.py]: OK, I renamed it, along with other similar tests.
